src/Cafe/OS/libs/nsyshid/BackendLibusb.cpp

```diff
@@ -680,32 +698,84 @@ namespace nsyshid::backend::libusb
 		return false;
 	}
 
-	bool DeviceLibusb::SetProtocol(uint32 ifIndex, uint32 protocol)
+	template<typename Configs, typename Function>
+	static int DoForEachInterface(const Configs& configs, uint8 config_num, Function action)
+	{
+		int ret = LIBUSB_ERROR_NOT_FOUND;
+		if (configs.size() <= config_num || !configs[config_num])
+			return ret;
+		for (uint8 i = 0; i < configs[config_num]->bNumInterfaces; ++i)
```

Comment:
See my other comment about invalid config descriptors above;
This looks like the place where having invalid config descriptors, that contain just a `nullptr`, will blow up.